### ISSUE TITLE
pip install zxing-cpp - needs git when building from source?

### ISSUE THREAD

**Issue by arechnitzer** [zxing-cpp/zxing-cpp#489]
I'm pip-installing zxing-cpp in a docker container and trying to keep things as minimal as possible.  

Can one pip-install without first installing git?

**Comment by cbm755** [comment]
note: Ubuntu 20.04 has Python 3.8 and there is no wheel for 3.8 so it builds from source (#488).


fails:
```
docker pull ubuntu:20.04
docker run -it --rm ubuntu:20.04
apt-get update
apt-get install python3-dev python3-pip
pip install zxing-cpp
```

works:
```
docker pull ubuntu:20.04
docker run -it --rm ubuntu:20.04
apt-get update
apt-get install python3-dev python3-pip
apt-get install git   # <--  seems related to pybind11?
pip install zxing-cpp
```

**Comment by axxel** [comment]
> Can one pip-install without first installing git?

Yes, if a prebuilt wheel is available. That is generally the case for the last 3 python versions (at the time of release). right now this would be 3.9, 3.10 and 3.11.

I [updated](https://github.com/zxing-cpp/zxing-cpp/commit/ca96dab34a293055378fc795fbe27abafa45fb3f) the readme to mention git.

> apt-get install git   # <--  seems related to pybind11?

It is also required to get the c++ sources of zxing-cpp itself.

**Comment by cbm755** [comment]
> It is also required to get the c++ sources of zxing-cpp itself.

That part doesn't make sense to me when installing a release...   Are the "c++ sources" not included in the .tar.gz from pypi?  viz, https://files.pythonhosted.org/packages/3d/19/a07c245a5603b5c84916066d782523697febbfe83daaee98e44b78813c24/zxing-cpp-2.0.0.tar.gz

Apologies for the naive questions, still learning the relationship between the Python library `zxing-cpp` and this (larger?) project.

**Comment by axxel** [comment]
> Are the "c++ sources" not included in the .tar.gz from pypi?

The (single) c++ source file of the python wrapper is included but not the core zxing-cpp library. The python module can be either build from the top-level cmake project where it would usually be dynamically linked to the c++ library or it can be built from the `setup.py` file as a 'standalone' (statically linked) python module. In the latter case, cmake determines whether the c++ sources are at `../../core`. If they are not, then we git-fetch them from github. Since the sources of pybind11 need to be fetched from there anyway, that does not make any practical difference.
